packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -580,6 +568,33 @@ export class DatabaseCluster extends DatabaseClusterNew {
       target: this,
     });
   }
+
+  private addSingleUserRotationForMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
+    if (!this.secret) {
+      throw new Error('Cannot add single user rotation for a cluster without master secret.');
+    }
+
+    const id = 'RotationSingleUser';
+    return this.addSingleUserRotationForSecret(this, id, this.secret, options);
+  }
+
+  private addSingleUserRotationForSecret(scope: Construct, id: string, secret: secretsmanager.ISecret, options: RotationSingleUserOptions)
+    : secretsmanager.SecretRotation {
+    if (Node.of(scope).tryFindChild(id)) {
+      const secretName = this === scope ? 'master' : 'additional'; // Is there a better way to do this?
+      throw new Error(`A single user rotation for ${secretName} secret was already added to this cluster.`);
```

Comment:
I think this error message is not good enough.

I understand it works for the master secret, but for a non-master secret, we need to specify it in more detail.

How about:

```ts
      const secretDescription = this === scope ? 'master cluster secret' : `secret '${secret.path}'`;
      throw new Error(`A single user rotation for ${secretDescription} was already added to this cluster`);
```

Reply:
I'm not an EN speaker, but "master cluster secret" sounds a bit confusing (the same for non-cluster; "main instance secret"). Let me know.

Reply:
Does "master secret of cluster/instance" sound better?

I'd rather not introduce a new terminology ("main") when we already have a different word used for that concept?

Reply:
Something like `Single user rotation was already enabled for the master secret of this cluster/instance` would probably work too.

Reply:
Done.